hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieClusteringConfig.java

```diff
@@ -0,0 +1,155 @@
+  public static final String DEFAULT_CLUSTERING_TARGET_PARTITIONS = String.valueOf(2);
+
+  // Each clustering operation can create multiple groups. Total amount of data processed by clustering operation
+  // is defined by below two properties (CLUSTERING_MAX_GROUP_SIZE * CLUSTERING_MAX_NUM_GROUPS).
+  // Max amount of data to be included in one group
+  public static final String CLUSTERING_MAX_GROUP_SIZE = "hoodie.clustering.max.group.size";
+  public static final String DEFAULT_CLUSTERING_MAX_GROUP_SIZE = String.valueOf(2 * 1024 * 1024 * 1024L);
+
+  public static final String CLUSTERING_MAX_NUM_GROUPS = "hoodie.clustering.max.num.groups";
+  public static final String DEFAULT_CLUSTERING_MAX_NUM_GROUPS = "1";
+
+  // Each group can produce 'N' (CLUSTERING_MAX_GROUP_SIZE/CLUSTERING_TARGET_FILE_SIZE) output file groups.
+  public static final String CLUSTERING_TARGET_FILE_SIZE = "hoodie.clustering.target.file.size";
+  public static final String DEFAULT_CLUSTERING_TARGET_FILE_SIZE = String.valueOf(1 * 1024 * 1024 * 1024L);
+
+  // Any strategy specific params can be saved with this prefix
+  public static final String CLUSTERING_STRATEGY_PARAM_PREFIX = "hoodie.clustering.strategy.param.";
+
+  // constants related to clustering that may be used by more than 1 strategy.
+  public static final String SORT_COLUMNS_PROPERTY = HoodieClusteringConfig.CLUSTERING_STRATEGY_PARAM_PREFIX + "sort.columns";
+
+  public HoodieClusteringConfig(Properties props) {
+    super(props);
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  public static class Builder {
+
+    private final Properties props = new Properties();
+
+    public Builder fromFile(File propertiesFile) throws IOException {
+      try (FileReader reader = new FileReader(propertiesFile)) {
+        this.props.load(reader);
+        return this;
+      }
+    }
+
+    public Builder withScheduleClusteringStrategyClass(String clusteringStrategyClass) {
+      props.setProperty(SCHEDULE_CLUSTERING_STRATEGY_CLASS, clusteringStrategyClass);
+      return this;
+    }
+
+    public Builder withRunClusteringStrategyClass(String runClusteringStrategyClass) {
+      props.setProperty(RUN_CLUSTERING_STRATEGY_CLASS, runClusteringStrategyClass);
+      return this;
+    }
+
+    public Builder withClusteringTargetPartitions(String clusteringTargetPartitions) {
+      props.setProperty(CLUSTERING_TARGET_PARTITIONS, clusteringTargetPartitions);
+      return this;
+    }
+
+    public Builder withClusteringMaxGroupSize(long clusteringMaxGroupSize) {
+      props.setProperty(CLUSTERING_MAX_GROUP_SIZE, String.valueOf(clusteringMaxGroupSize));
+      return this;
+    }
+
+    public Builder withClusteringMaxNumGroups(int maxNumGroups) {
+      props.setProperty(CLUSTERING_MAX_NUM_GROUPS, String.valueOf(maxNumGroups));
+      return this;
+    }
+
+    public Builder withClusteringTargetFileSize(long targetFileSize) {
```

Comment:
Same this, if in MB, add it to the name ?

Reply:
Fixed